src/util/checkEnvVariables.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { expectedChainVariables, expectedCommonVariables } from '../constants';
+
+type ProcessEnv = string | undefined;
+
+const checkEnvVariable = (variable: string) => {
+  const envVariable = process.env[variable]!;
+  let found = false;
+
+  expectedChainVariables.map((chainVariable) =>
+    [1, 2].map((chainNumber) => {
+      const envVariablePerChain: ProcessEnv = process.env[`REACT_APP_${chainVariable}_CHAIN_${chainNumber}`];
+      if (envVariablePerChain) {
+        found = true;
+      }
+    })
+  );
+
+  [1, 2].map((chainNumber) => {
+    const envVariablePerChain: ProcessEnv = process.env[`REACT_APP_CHAIN_${chainNumber}`];
+    if (envVariablePerChain) {
+      found = true;
+    }
+  });
+
+  expectedCommonVariables.map((chainVariable) => {
+    const envVariable: ProcessEnv = process.env[`REACT_APP_${chainVariable}`];
+    if (envVariable) {
+      found = true;
+    }
+  });
+
+  if (!found) {
+    throw new Error(`Env Variable ${variable} was not defined`);
+  }
```

Comment:
Wait, this thing is mixing functionality quite heavily. We do two things here:

1. Extract a requested environment variable (`variable` argument) and fail if it's not set.
2. Validate that some more expected variables are being set.

That should be split into two separate functions, where (2) is simply being called at the application startup and not for every unrelated usage of `checkEvnVariable` function.